### PR TITLE
refactor: clean up some compiler warnings

### PR DESCRIPTION
**Pull request by gocarlos**

### CODE REVIEW

File: test/mbus_parse_hex.c [hunk 1]
```diff
@@ -48,11 +48,12 @@ main(int argc, char *argv[])
     }
 
     memset(raw_buff, 0, sizeof(raw_buff));
-    len = fread(raw_buff, 1, sizeof(raw_buff), fp);
+    fread(raw_buff, 1, sizeof(raw_buff), fp);
 
     if (ferror(fp) != 0)
     {
         fprintf(stderr, "%s: failed to read '%s'\n", argv[0], file);
+        fclose(fp);
```
Comment (gocarlos) [inline test/mbus_parse_hex.c]: leak

File: test/mbus_parse.c [hunk 2]
```diff
@@ -53,6 +53,7 @@ main(int argc, char *argv[])
     if (ferror(fp) != 0)
     {
         fprintf(stderr, "%s: failed to read '%s'\n", argv[0], file);
+        fclose(fp);
```
Comment (gocarlos) [inline test/mbus_parse.c]: leak

File: mbus/mbus-protocol.c [hunk 3]
```diff
@@ -3901,7 +3901,7 @@ mbus_data_variable_header_xml(mbus_data_variable_header *header)
 /// Generate XML for a single variable-length data record
 //------------------------------------------------------------------------------
 char *
-mbus_data_variable_record_xml(mbus_data_record *record, int record_cnt, int frame_cnt, mbus_data_variable_header *header)
+mbus_data_variable_record_xml(mbus_data_record *record, int record_cnt, int frame_cnt)
```
Comment (gocarlos) [inline mbus/mbus-protocol.c]: used variables in function

File: mbus/mbus-protocol-aux.c [hunk 4]
```diff
@@ -785,8 +785,6 @@ mbus_register_found_event(mbus_handle * handle, void (*event)(mbus_handle * hand
 
 int mbus_fixed_normalize(int medium_unit, long medium_value, char **unit_out, double *value_out, char **quantity_out)
 {
-    double exponent = 0.0;
```
Comment (gocarlos) [inline mbus/mbus-protocol-aux.c]: used vars

File: mbus/mbus-protocol.c (hunk 3 above)
Comment (lategoodbye) [reply mbus/mbus-protocol.c]: I understand this is unused, but i would like to keep it as long these functions aren't fully implemented yet.

**Comment by lategoodbye** [comment]
Thanks

**Pull request merged**